packages/dev/test/RelayHubGasCalculations.test.ts

```diff
@@ -464,114 +472,127 @@ contract('RelayHub gas calculations', function ([_, relayOwner, relayWorker, rel
   })
 
   describe('check calculation does not break for different fees', function () {
-    before(async function () {
-      await relayHub.depositFor(relayOwner, { value: (1).toString() })
-    });
-
     [0, 1000]
       .forEach(messageLength =>
-        [0, 1, 100]
-          .forEach(requestedFee => {
-            // avoid duplicate coverage checks. they do the same, and take a lot of time:
-            if (requestedFee !== 0 && messageLength !== 0 && process.env.MODE === 'coverage') return
-            // 50k tests take more than 10 seconds to complete so will run once for sanity
-            if (messageLength === 50000 && requestedFee !== 10) return
-            it(`should compensate relay with requested fee of ${requestedFee.toString()}% with ${messageLength.toString()} calldata size`,
-              async function () {
-                const beforeBalances = await getBalances()
-                const pctRelayFee = requestedFee.toString()
-                const senderNonce = (await forwarderInstance.getNonce(senderAddress)).toString()
-                const encodedFunction = recipient.contract.methods.emitMessage('a'.repeat(messageLength)).encodeABI()
-                const baseRelayFee = '0'
-                const relayRequest: RelayRequest = {
-                  request: {
-                    to: recipient.address,
-                    data: encodedFunction,
-                    from: senderAddress,
-                    nonce: senderNonce,
-                    value: '0',
-                    gas: gasLimit.toString(),
-                    validUntil: '0'
-                  },
-                  relayData: {
-                    baseRelayFee,
-                    pctRelayFee,
-                    transactionCalldataGasUsed: '',
-                    maxFeePerGas: maxFeePerGas.toString(),
-                    maxPriorityFeePerGas: maxPriorityFeePerGas.toString(),
-                    relayWorker,
+        [0, 10, 100]
+          .forEach(requestedFee =>
+            [0, 20].forEach(devFee => {
+              // avoid duplicate coverage checks. they do the same, and take a lot of time:
+              if (requestedFee !== 0 && messageLength !== 0 && process.env.MODE === 'coverage') return
+              // 50k tests take more than 10 seconds to complete so will run once for sanity
+              if (messageLength === 50000 && requestedFee !== 10) return
+              it(
+                `should compensate relay with requested fee of ${requestedFee.toString()}%, dev fee of ${devFee.toString()}% and ${messageLength.toString()} calldata size`,
+                async function () {
+                  let gasOverhead = parseInt(defaultEnvironment.relayHubConfiguration.gasOverhead.toString())
+                  if (devFee !== 0) {
```

Comment:
huh? read doesn't matter for zero/nonzero

Reply:
It also includes a 'write'

Reply:
hard-coding "2800" this way means it going to cause headache when moved to different networks (e.g. arbitrum)

Reply:
?

Reply:
Fixed